tor/helpers/flair.py

```diff
@@ -39,12 +41,32 @@ def update_user_flair(post, config):
     On a successful transcription, this takes the user's current flair,
```

Comment:
Oh — This function is also way too long and complex, I'd try to break this into several other bits.

Reply:
Generally I try to break things up if they meet one of the following criteria:

* over 500 lines long
* has sections that are untestable
* is objectively unreadable without further abstraction

This doesn't hit any of those criteria; why do you think this is too long and complex?

Reply:
37 lines of code makes Codacy complain :)

It would also reduce the block nesting, which would be good.

Reply:
...bah.

Reply:
my bad, I meant codebeat